### Business Questions:

1. What are the most common customer request categories?
2. Which ticket types have the highest priority?
3. Which categories may be suitable for automation?
4. Which channels generate the most support requests?
5. Which categories are associated with low customer satisfaction?
6. What percentage of tickets could potentially be handled automatically?

### Dataset 1

Analysing this dataset to see how it would able to answer those Business Questions and find that which columns would be usefull in Automation that we eager to create.

In [48]:
import pandas as pd
import numpy as np
from scipy.stats import chi2_contingency

In [3]:
df = pd.read_csv(r"..\data\processed\tickets.csv")

In [4]:
df.head()

,subject,message,response,type,category,priority,language,tag
0,Wesentlicher Sicherheitsvorfall,"Sehr geehrtes Support-Team,\n\nich möchte eine...",Vielen Dank für die Meldung des kritischen Sic...,Incident,Technical Support,high,de,Security
1,Account Disruption,"Dear Customer Support Team,\n\nI am writing to...","Thank you for reaching out, <name>. We are awa...",Incident,Technical Support,high,en,Account
2,Query About Smart Home System Integration Feat...,"Dear Customer Support Team,\n\nI hope this mes...",Thank you for your inquiry. Our products suppo...,Request,Returns and Exchanges,medium,en,Product
3,Inquiry Regarding Invoice Details,"Dear Customer Support Team,\n\nI hope this mes...",We appreciate you reaching out with your billi...,Request,Billing and Payments,low,en,Billing
4,Question About Marketing Agency Software Compa...,"Dear Support Team,\n\nI hope this message reac...",Thank you for your inquiry. Our product suppor...,Problem,Sales and Pre-Sales,medium,en,Product


In [5]:
df.columns

Index(['subject', 'message', 'response', 'type', 'category', 'priority',
       'language', 'tag'],
      dtype='str')

#### Category Analysis

In [6]:
df["category"].value_counts(normalize=True).mul(100).round(2).astype(str) + "%"

category
Technical Support                  29.25%
Product Support                    18.37%
Customer Service                   14.93%
IT Support                         12.01%
Billing and Payments                9.75%
Returns and Exchanges               5.03%
Service Outages and Maintenance     4.02%
Sales and Pre-Sales                 3.21%
Human Resources                     2.01%
General Inquiry                     1.42%
Name: proportion, dtype: str

In [28]:
pd.crosstab(df["category"], df["priority"], 
            normalize="index").mul(100).round(2).astype(str) + "%" 

priority,high,low,medium
category,,,
Billing and Payments,30.56%,21.59%,47.85%
Customer Service,18.86%,35.07%,46.06%
General Inquiry,12.84%,55.56%,31.6%
Human Resources,9.55%,49.65%,40.8%
IT Support,48.85%,8.88%,42.27%
Product Support,29.59%,20.14%,50.27%
Returns and Exchanges,21.36%,36.95%,41.68%
Sales and Pre-Sales,17.54%,36.17%,46.3%
Service Outages and Maintenance,70.99%,11.5%,17.51%


In [29]:
pd.crosstab(df["category"], df["language"],
            normalize="index").mul(100).round(2).astype(str) + "%"

language,de,en
category,,
Billing and Payments,42.79%,57.21%
Customer Service,43.53%,56.47%
General Inquiry,41.73%,58.27%
Human Resources,39.58%,60.42%
IT Support,43.43%,56.57%
Product Support,41.49%,58.51%
Returns and Exchanges,42.94%,57.06%
Sales and Pre-Sales,44.12%,55.88%
Service Outages and Maintenance,42.16%,57.84%


In [53]:
table = pd.crosstab(df["category"], df["priority"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print("p-value: ", p_value)
print("Degrees of freedom: ", dof)

Chi-square: 4667.36
p-value:  0.0
Degrees of freedom:  18


In [54]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is a statistically significant relationship between category and priority.


In [55]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.29


The Chi-square test showed a statistically significant association between ticket category and priority, as the p-value was far below 0.05. The distribution of priority levels is not the same across all categories; for example, Service Outages and Maintenance and Technical Support tickets have higher proportions of high-priority cases. Cramér’s V was 0.29, indicating a moderate association between category and priority.

#### Priority Analysis

In [30]:
df["priority"].value_counts(normalize=True).mul(100).round(2).astype(str) + "%"

priority
medium    40.28%
high       39.1%
low       20.62%
Name: proportion, dtype: str

#### Language Analysis

In [41]:
pd.crosstab(df["priority"], df["language"])

language,de,en
priority,,
high,4832,6346
low,2520,3374
medium,4897,6618


In [ ]:
pd.crosstab(df["priority"], df["language"],
            normalize="index").mul(100).round(2).astype(str) + "%"

language,de,en
priority,,
high,43.23%,56.77%
low,42.76%,57.24%
medium,42.53%,57.47%


In [56]:
table = pd.crosstab(df["priority"], df["language"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom: ", dof)

Chi-square: 1.16
p-value: 0.56
Degrees of freedom:  2


In [57]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [58]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.01


The Chi-square test found no statistically significant association between ticket priority and language, as the p-value was greater than 0.05. Therefore, the distribution of priority levels appears to be similar across different languages. Additionally, the low Cramér’s V value indicates that the relationship between language and priority is weak.

#### Response Analysis

In [59]:
table = pd.crosstab(df["category"], df["response"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom: ", dof)

Chi-square: 257220.00
p-value: 0.49
Degrees of freedom:  257211


In [64]:
df["response"].nunique()

28580

In [63]:
print((expected < 5).mean())

1.0


The chi-square is not reliable for this relation.

In [60]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [61]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 1.00


The Chi-square test and Cramér’s V are not appropriate for examining the relationship between `category` and `response` in this dataset. The `response` column contains free-text values with a very high number of unique responses, resulting in a large and sparse contingency table. Therefore, the non-significant p-value and Cramér’s V of 1.00 should not be interpreted as evidence of either no relationship or a perfect relationship. Instead, responses should first be converted into meaningful groups, such as topics, intents, or sentiment categories, before performing the analysis.

Next we want to see if there is similar responses to similar requests or not.

In [67]:
df[df["category"] == "Technical Support"][["message", "response"]].sample(5)

,message,response
20450,The financial firm's data analytics tools are ...,We will assist in resolving the data analytics...
13928,Seeking information on digital marketing tacti...,Appreciate your inquiry. We provide a range of...
16981,Marketing campaigns are underperforming due to...,Investigate the technical issue promptly
763,"Dear Customer Support Team,\n\nI hope this mes...",Thank you for reaching out regarding the probl...
25931,Marketing Agency Experiencing Malfunctions wit...,Please contact us to discuss the functionality...


In [71]:
df[df["category"] == "Service Outages and Maintenance"][["message", "response"]].sample(5)

,message,response
12717,Could you please give us more information abou...,"<name>, we appreciate your interest in DataRob..."
15753,There was an unexpected outage that affected u...,<name> acknowledges the SaaS platform access i...
25017,The agency faced service disruptions impacting...,We apologize for the disruptions with digital ...
27018,"customer support, expressing concern about the...","<name>, we take data security seriously and re..."
7158,"Customer Support, I am submitting a request to...","<name>, thank you for your email regarding the..."


In [73]:
df[df["category"] == "Customer Service"][["message", "response"]].sample(5)

,message,response
2336,Ein Sicherheitsvorfall könnte möglicherweise d...,"Sehr geehrte/r <name>, ich bestätige den Erhal..."
20078,"Dear Customer Service, I am writing to request...","Dear <name>, thank you for your message. We ha..."
13994,Is it possible to get information on securing ...,Offer general guidance on securing medical dat...
25991,Benutzer haben Schwierigkeiten bei der Synchro...,Wir werden das Synchronisationsproblem in der ...
4749,Unregelmäßigkeiten in den Daten beeinträchtige...,"<name>, wir entschuldigen die Unstimmigkeiten ..."


### Dataset 2

In [128]:
df = pd.read_csv(r"..\data\processed\customer_support.csv")

In [129]:
df.head()

,ticket_id,customer_name,customer_email,customer_age,customer_gender,product_purchased,purchase_date,ticket_type,subject,message,status,priority,channel,customer_satisfaction_rating
0,1,Marisa Obrien,carrollallison@example.com,32,Other,GoPro Hero,2021-03-22,Technical issue,Product setup,I'm having an issue with the {product_purchase...,Pending Customer Response,Critical,Social media,NaN
1,2,Jessica Rios,clarkeashley@example.com,42,Female,LG Smart TV,2021-05-22,Technical issue,Peripheral compatibility,I'm having an issue with the {product_purchase...,Pending Customer Response,Critical,Chat,NaN
2,3,Christopher Robbins,gonzalestracy@example.com,48,Other,Dell XPS,2020-07-14,Technical issue,Network problem,I'm facing a problem with my {product_purchase...,Closed,Low,Social media,3.0
3,4,Christina Dillon,bradleyolson@example.org,27,Female,Microsoft Office,2020-11-13,Billing inquiry,Account access,I'm having an issue with the {product_purchase...,Closed,Low,Social media,3.0
4,5,Alexander Carroll,bradleymark@example.com,67,Female,Autodesk AutoCAD,2020-02-04,Billing inquiry,Data loss,I'm having an issue with the {product_purchase...,Closed,Low,Email,1.0


In [130]:
df.columns

Index(['ticket_id', 'customer_name', 'customer_email', 'customer_age',
       'customer_gender', 'product_purchased', 'purchase_date', 'ticket_type',
       'subject', 'message', 'status', 'priority', 'channel',
       'customer_satisfaction_rating'],
      dtype='str')

#### Customer Support Analysis

In [131]:
df["subject"].value_counts(normalize=True).mul(100).round(2).astype(str) + "%"

subject
Refund request               6.8%
Software bug                6.78%
Product compatibility        6.7%
Delivery problem            6.62%
Hardware issue              6.46%
Battery life                 6.4%
Network problem             6.36%
Installation support        6.26%
Product setup               6.25%
Payment issue               6.21%
Product recommendation       6.1%
Account access              6.01%
Peripheral compatibility    5.86%
Data loss                    5.8%
Cancellation request        5.75%
Display issue               5.64%
Name: proportion, dtype: str

In [132]:
df["ticket_type"].value_counts(normalize=True).mul(100).round(2).astype(str) + "%"

ticket_type
Refund request          20.69%
Technical issue         20.63%
Cancellation request    20.01%
Product inquiry         19.38%
Billing inquiry         19.29%
Name: proportion, dtype: str

In [78]:
pd.crosstab(df["subject"], df["priority"],
            normalize="index").mul(100).round(2).astype(str) + "%"

priority,Critical,High,Low,Medium
subject,,,,
Account access,22.2%,25.54%,25.74%,26.52%
Battery life,22.88%,24.72%,24.54%,27.86%
Cancellation request,25.87%,24.23%,24.44%,25.46%
Data loss,26.07%,26.07%,22.4%,25.46%
Delivery problem,22.64%,26.74%,22.82%,27.81%
Display issue,25.1%,25.31%,23.43%,26.15%
Hardware issue,28.34%,23.77%,27.79%,20.11%
Installation support,24.34%,26.79%,24.34%,24.53%
Network problem,26.35%,24.3%,24.12%,25.23%


In [84]:
table = pd.crosstab(df["subject"], df["priority"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom: ", dof)

Chi-square: 40.86
p-value: 0.65
Degrees of freedom:  45


In [85]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [86]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.04


The Chi-square test found no statistically significant association between `subject` and `priority` ((chi2(45) = 40.86\), \(p = 0.648\)). Additionally, Cramér’s V was 0.04, indicating a very weak relationship between the two variables. Therefore, `subject` does not appear to be meaningfully associated with ticket priority in this dataset.

In [87]:
pd.crosstab(df["ticket_type"], df["priority"],
            normalize="index").mul(100).round(2).astype(str) + "%"

priority,Critical,High,Low,Medium
ticket_type,,,,
Billing inquiry,25.7%,23.38%,24.36%,26.56%
Cancellation request,24.96%,23.48%,24.42%,27.14%
Product inquiry,24.56%,24.31%,24.25%,26.87%
Refund request,25.34%,25.57%,25.11%,23.97%
Technical issue,25.13%,26.22%,23.64%,25.01%


In [91]:
table = pd.crosstab(df["ticket_type"], df["priority"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom:", dof)

Chi-square: 10.63
p-value: 0.56
Degrees of freedom: 12


In [92]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [93]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.02


The Chi-square test found no statistically significant association between `ticket_type` and `priority` ((chi2(12) = 10.63\), \(p = 0.56\)). Additionally, Cramér’s V was 0.02, indicating an extremely weak relationship between the two variables. Therefore, `ticket_type` does not appear to be meaningfully associated with ticket priority in this dataset.

#### Satisfaction Analysis

In [100]:
pd.crosstab(df["ticket_type"], df["customer_satisfaction_rating"],
            normalize="index").mul(100).round(2).astype(str) + "%"

customer_satisfaction_rating,1.0,2.0,3.0,4.0,5.0
ticket_type,,,,,
Billing inquiry,18.01%,22.98%,17.28%,21.69%,20.04%
Cancellation request,20.54%,17.83%,21.51%,18.41%,21.71%
Product inquiry,18.57%,18.76%,22.89%,21.95%,17.82%
Refund request,22.15%,19.46%,20.47%,18.62%,19.3%
Technical issue,20.34%,20.0%,22.59%,17.59%,19.48%


In [101]:
table = pd.crosstab(df["ticket_type"], df["customer_satisfaction_rating"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom:", dof)

Chi-square: 19.24
p-value: 0.26
Degrees of freedom: 16


In [102]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [103]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.04


The Chi-square test found no statistically significant association between `ticket_type` and `customer_satisfaction_rating` ((chi2(16) = 19.24\), \(p = 0.26\)). Additionally, Cramér’s V was 0.04, indicating a very weak relationship between the two variables. Therefore, `ticket_type` does not appear to be meaningfully associated with customer satisfaction ratings in this dataset.

In [112]:
pd.crosstab(df["subject"], df["customer_satisfaction_rating"],
            normalize="index").mul(100).round(2).astype(str) + "%"

customer_satisfaction_rating,1.0,2.0,3.0,4.0,5.0
subject,,,,,
Account access,22.81%,20.47%,18.71%,15.79%,22.22%
Battery life,21.39%,21.39%,17.34%,20.23%,19.65%
Cancellation request,17.57%,19.59%,22.3%,19.59%,20.95%
Data loss,26.53%,13.61%,19.05%,18.37%,22.45%
Delivery problem,23.6%,24.72%,19.1%,12.92%,19.66%
Display issue,18.71%,17.42%,21.94%,21.29%,20.65%
Hardware issue,18.03%,20.77%,22.4%,20.77%,18.03%
Installation support,13.92%,19.62%,24.05%,18.99%,23.42%
Network problem,22.39%,21.39%,18.41%,21.39%,16.42%


In [113]:
table = pd.crosstab(df["subject"], df["customer_satisfaction_rating"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom:", dof)

Chi-square: 47.65
p-value: 0.88
Degrees of freedom: 60


In [114]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [115]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.07


The Chi-square test found no statistically significant association between `subject` and `customer_satisfaction_rating` ((chi2(60) = 47.65\), \(p = 0.88\)). Additionally, Cramér’s V was 0.07, indicating a very weak relationship between the two variables. Therefore, `subject` does not appear to be meaningfully associated with customer satisfaction ratings in this dataset.

In [117]:
pd.crosstab(df["priority"], df["customer_satisfaction_rating"],
            normalize="index").mul(100).round(2).astype(str) + "%"

customer_satisfaction_rating,1.0,2.0,3.0,4.0,5.0
priority,,,,,
Critical,20.94%,20.11%,20.52%,19.01%,19.42%
High,20.85%,20.14%,19.86%,18.16%,20.99%
Low,18.48%,19.72%,19.25%,23.14%,19.41%
Medium,19.45%,19.31%,24.06%,18.44%,18.73%


In [118]:
table = pd.crosstab(df["priority"], df["customer_satisfaction_rating"])

chi2, p_value, dof, expected = chi2_contingency(table)

print(f"Chi-square: {chi2:.2f}")
print(f"p-value: {p_value:.2f}")
print("Degrees of freedom:", dof)

Chi-square: 12.59
p-value: 0.40
Degrees of freedom: 12


In [119]:
if p_value < 0.05:
    print("There is a statistically significant relationship between category and priority.")
else:
    print("There is insufficient statistical evidence for a significant association.")

There is insufficient statistical evidence for a significant association.


In [120]:
n = table.to_numpy().sum()
r, k = table.shape

cramers_v = np.sqrt(
    (chi2 / n) / min(r - 1, k - 1)
)

print(f"Cramer's V: {cramers_v:.2f}")

Cramer's V: 0.04


The Chi-square test found no statistically significant association between `priority` and `customer_satisfaction_rating` ((chi2(12) = 12.59\), \(p = 0.40\)). Additionally, Cramér’s V was 0.04, indicating a very weak relationship between the two variables. Therefore, ticket priority does not appear to be meaningfully associated with customer satisfaction ratings in this dataset.

### Dataset 3

In [134]:
df = pd.read_csv(r"..\data\processed\FAQ.csv")

In [135]:
df.head()

,document_id,question,answer
0,faq_1,How can I create an account?,"To create an account, click on the 'Sign Up' b..."
1,faq_2,What payment methods do you accept?,"We accept major credit cards, debit cards, and..."
2,faq_3,How can I track my order?,You can track your order by logging into your ...
3,faq_4,What is your return policy?,Our return policy allows you to return product...
4,faq_5,Can I cancel my order?,You can cancel your order if it has not been s...


In [136]:
df["question"].head(20)

0                          How can I create an account?
1                   What payment methods do you accept?
2                             How can I track my order?
3                           What is your return policy?
4                                Can I cancel my order?
5                          How long does shipping take?
6                  Do you offer international shipping?
7     What should I do if my package is lost or dama...
8     Can I change my shipping address after placing...
9                   How can I contact customer support?
10                 Do you offer gift wrapping services?
11                  What is your price matching policy?
12                                Can I order by phone?
13          Are my personal and payment details secure?
14                What is your price adjustment policy?
15                       Do you have a loyalty program?
16             Can I order without creating an account?
17            Do you offer bulk or wholesale dis

In [139]:
df["question"].sample(20)

12                                Can I order by phone?
8     Can I change my shipping address after placing...
46    Can I request a product repair or replacement ...
58    Can I request a product if it is listed as 'co...
69    Can I return a product if it was damaged due t...
28    What should I do if my discount code is not wo...
40    Can I request a product that is currently out ...
9                   How can I contact customer support?
47    Can I order a product if it is listed as 'out ...
1                   What payment methods do you accept?
7     What should I do if my package is lost or dama...
68    Can I order a product if it is listed as 'comi...
2                             How can I track my order?
22                     Do you offer expedited shipping?
10                 Do you offer gift wrapping services?
3                           What is your return policy?
17            Do you offer bulk or wholesale discounts?
52    Can I request a product if it is not curre

#### Knowledge Base Coverage Analysis

| Ticket Domain       | Ticket Evidence                         | FAQ Coverage                     | وضعیت      |
| ------------------- | --------------------------------------- | -------------------------------- | -----------|
| Billing / Payment   | 9.75% Category + 19.29% Billing Inquiry | Payment Methods                  | 🟢 Strong  |
| Returns / Refund    | 5.03% Category + 20.69% Refund Request  | Return Policy                    | 🟢 Strong  |
| Cancellation        | 20.01% Ticket Type + 5.75% Subject      | Cancel Order                     | 🟢 Strong  |
| Shipping / Delivery | Delivery Problem 6.62%                  | Tracking, Shipping, Lost Package | 🟢 Strong  |
| Account             | Account Access 6.01%                    | Account Creation                 | 🟡 Partial |
| Sales / Pre-Sales   | 3.21%                                   | Discounts, Price Matching        | 🟡 Partial |
| Product Inquiry     | 19.38%                                  | Limited Product FAQs             | 🟡 Partial |
| Technical Support   | 29.25%                                  | No Technical FAQ                 | 🔴 Weak    |
| IT Support          | 12.01%                                  | No Direct FAQ                    | 🔴 Weak    |
| Service Outages     | 4.02%                                   | No Direct FAQ                    | 🔴 Weak    |
| Human Resources     | 2.01%                                   | No Direct FAQ                    | 🔴 Weak    |
| General Inquiry     | 1.42%                                   | Partial                          | 🟡 Partial |


In [143]:
faq_topics = {
    "account": [
        "How can I create an account?",
        "Can I order without creating an account?"
    ],
    "payment": ["What payment methods do you accept?"],
    "order": ["Can I order without creating an account?"],
    "order_tracking": ["How can I track my order?"],
    "returns": ["What is your return policy?"],
    "cancellation": [
        "Can I cancel my order?",
        "Can I change or cancel an item in my order?"
    ],
    "shipping": [
        "How long does shipping take?",
        "Do you offer international shipping?",
        "What should I do if my package is lost or damaged?",
        "Can I change my shipping address after placing an order?"
    ],
    "customer_service": [" How can I contact customer support?"],
    "pricing": [
       "What is your price matching policy?",
        "What is your price adjustment policy?"
        ],
    "sales": [
        "Do you offer bulk or wholesale discounts?",
        "Do you offer gift wrapping services?"
    ],
    "security": ["Are my personal and payment details secure?"]
}

In [ ]:
ticket_topics = {
    "payment": ["Payment issue"],
    "returns": ["Refund request"],
    "cancellation": ["Cancellation request"],
    "shipping": ["Delivery problem"],
    "account": ["Account access"],
    "technical": [
        "Software bug",
        "Hardware issue",
        "Network problem"
    ]
}

This is just an example of waht we can have from FAQ dataset to Automate the customer requests.